src/codegen/doc/format.rs

```diff
@@ -54,121 +57,258 @@ fn get_language<'a>(entry: &'a str, out: &mut String) -> &'a str {
     entry
 }
 
-fn format(mut input: &str, symbols: &symbols::Info, in_type: &str) -> String {
+fn format(input: &str, env: &Env, in_type: &str) -> String {
     let mut ret = String::with_capacity(input.len());
-    loop {
-        let (before, after) = try_split(input, "`");
-        ret.push_str(&replace_c_types(before, symbols, in_type));
-        if let Some(after) = after {
-            ret.push('`');
-            let (before, after) = try_split(after, "`");
-            // don't touch anything enclosed in backticks
-            ret.push_str(before);
-            if let Some(after) = after {
-                ret.push('`');
-                input = after;
-            } else {
-                return ret;
-            }
-        } else {
-            return ret;
-        }
-    }
+    // We run gi_docgen first because it's super picky about the types it replaces
+    let no_c_types_re = gi_docgen::replace_c_types(input, env, in_type);
+    ret.push_str(&replace_c_types(&no_c_types_re, env, in_type));
+    ret
 }
 
-static SYMBOL: Lazy<Regex> =
-    Lazy::new(|| Regex::new(r"(^|[^\\])([@#%])(\w+\b)([:.]+[\w-]+\b)?").unwrap());
+static SYMBOL: Lazy<Regex> = Lazy::new(|| Regex::new(r"([#%])(\w+\b)([:.]+[\w-]+\b)?").unwrap());
+static PARAM_SYMBOL: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"([@])(\w+\b)([:.]+[\w-]+\b)?").unwrap());
 static FUNCTION: Lazy<Regex> =
     Lazy::new(|| Regex::new(r"([@#%])?(\w+\b[:.]+)?(\b[a-z0-9_]+)\(\)").unwrap());
-static GDK_GTK: Lazy<Regex> = Lazy::new(|| Regex::new(r"G[dt]k[A-Z]\w+\b").unwrap());
+// **note**
+// The optional . at the end is to make the regex more relaxed for some weird broken cases on gtk3's docs
+// it doesn't hurt other docs so please don't drop it
```

Comment:
Have an example here? Would be nice to have it in the code at least (or a test) so that future developers know what to keep in mind.

Reply:
There were some occurrences in gtk with `GtkSomething.` and it wasn't picked, although, that would be a bug to fix in the docs... I don't really mind dropping it :) it was mostly as an explanation for the review phase